site/app/libraries/MultiIterator.php

```diff
@@ -2,27 +2,38 @@
 
 namespace app\libraries;
 
+use app\exceptions\NotImplementedException;
+
 /**
  * Class MultiIterator
  * @package app\libraries
  *
- * Allows the chaining of multiple iterators in one iterator object
+ * An iterator that allows for sequentially iterating over each passed
+ * iterator in-order, one at a time. This means that iterator2 will not be
+ * called until iterator1 has been exhausted. This differs from
+ * MultipleIterator as that commits all iterator functions over all attached
+ * iterators in parallel (i.e. calling next() will return an array with the
+ * results of next() on all attached iterators).
+ *
+ * @see \MultipleIterator
  */
 class MultiIterator implements \Iterator {
```

Comment:
I think it make sense to rename `MultiIterator` to `CascadingIterator` or `SerialIterator` so its more descriptive and less similar sounding to `MultipleIterator`.